dbt/include/clickhouse/macros/utils/utils.sql

```diff
@@ -63,11 +63,15 @@
 {% endmacro %}
 
 
-{% macro clickhouse__listagg(measure, delimiter_text, order_by_clause, limit_num) %}
-  {{ exceptions.raise_compiler_error(
-    'ClickHouse does not support the listagg function.  See the groupArray function instead')
-    }}
-{% endmacro %}
+{% macro clickhouse__listagg(measure, delimiter_text, order_by_clause, limit_num) -%}
+    {% set arr = "array_agg({})".format(measure) %}
+    {% set arr = "arraySort({}, {})".format(arr, order_by_clause) if order_by_clause else arr %}
```

Comment:
The order by part doesn't work. You can validate the entire macro with the test's result.

Reply:
Thanks again. Now, I'm not sure how I checked this in the first place

I've considered two options here:
```
arraySlice(arrayMap(x -> (x.1), arraySort(x -> (x.2), arrayZip(array_agg(a), array_agg(d)))), 1, 10)

<Debug> executeQuery: Read 50000000 rows, 238.42 MiB in 8.861773 sec., 5642211.778613603 rows/sec., 26.90 MiB/sec.
<Debug> MemoryTracker: Peak memory usage (for query): 1.23 GiB.
```

and 

```
select  arraySlice(mapValues(mapSort(mapFromArrays(array_agg(d), array_agg(a)))), 1,10) from t;

<Debug> executeQuery: Read 50000000 rows, 238.42 MiB in 15.139353 sec., 3302651.0445988015 rows/sec., 15.75 MiB/sec.
<Debug> MemoryTracker: Peak memory usage (for query): 1.23 GiB.
```
Opted for the first one with the results